### ISSUE TITLE
Test Case 10: Time Limit Exceeded

### ISSUE THREAD

**Issue by stxe09** [nus-cs2030/2223-s2#206]
# Summary

![image](https://user-images.githubusercontent.com/122586851/220288514-f30d1ab3-eafc-4a0a-90f0-1dd5d669c88c.png)

Anyone faced the same issue? There's no output so I don't know what to change...

**Comment by xyxyiuxy** [comment]
If you have an intermediate class after WaitEvent, you may check whether you have created an infinite loop, can use System.out.println to check how many times you have actually used the getServiceTime() and check whether u update the server list correctly. Hope it can help you!

**Comment by jianjiezzz** [comment]
I think that u did not update correctly ur server's available time after serve event, therefore resulting in your wait event to keep returning another wait event.

**Comment by andersooi** [comment]
I faced the same problem as well and used a bunch of `System.out.println`s in my Simulator class to find out where the infinite loop was.

**Comment by stxe09** [comment]
> I faced the same problem as well and used a bunch of `System.out.println`s in my Simulator class to find out where the infinite loop was.

I tried that but my jshell just hangs and nothing gets printed.

**Comment by andersooi** [comment]
Yeah, it took me a while to experiment where I should put the print statements. Eventually, I found that placing it at the start of my while loop + end and having a few in between helped. Not sure if it will work for you, but you may also have to experiment on the placement of the print statements.

**Comment by wmingchuan** [comment]
Might be either the server's timings not updated or the queue not updated. Usually this is caused by some kind of infinite loop. Use a print statement and try to find out where the loop is. Use control C to stop the program.
